src/security_utilities_rust/src/microsoft_security_utilities_core/identifiable_scans.rs

```diff
@@ -1117,50 +1103,52 @@ impl Scan {
                 break;
             }
 
-            self.concurrent_parse_bytes(state, &buf[..len]);
+            self.parse_bytes(state, &buf[..len]);
         }
 
         /* Scanned all blocks */
         Ok(())
     }
+}
+
+pub struct Scan {
+    engine: ScanEngine,
+    state: ScanState,
+}
+
+impl Scan {
+    pub fn new(options: ScanOptions) -> Self {
+        Self {
+            engine: ScanEngine::new(options),
+            state: ScanState::default(),
+        }
```

Comment:
@beaubelgrave This is a breaking change for existing customers right? Now users need to access the checks via the state. I will add a release note entry in this PR to highlight this breaking change.